AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLDocument/GraphQLSubscription.swift

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+
+/// A concrete implementation of `SingleDirectiveGraphQLDocument` that represents a subscription operation.
+public struct GraphQLSubscription: SingleDirectiveGraphQLDocument {
+
+    public init(operationType: GraphQLOperationType,
+                name: String,
+                inputs: [GraphQLParameterName: GraphQLDocumentInput],
+                selectionSet: SelectionSet?) {
+        self.operationType = operationType
```

Comment:
here and above: do we need to init with an OperationType if we're creating a GraphQLSubscription ? Wouldn't the OperationType be inferrable in this init?